src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtilsUnitTest.java

```diff
@@ -970,9 +974,31 @@ public void testConstructPhaseGroups(final List<VariantContext> calls,
         spandelCalls.add(spannedSnpVC);
 
         tests.add(new Object[]{spandelCalls, new HashMap<>(spanDelHapMap), 2, 2, 1, 1, 1});
+
+        // test 11: a hom followed by two opposite-phase hets
```

Comment:
I understand that by virtue of the `for (i = 0; i < numVariants; i++) { . . . for (j = i + 1; j < numVariants; j++ . . .` nesting the case where the hom precedes the hets is the thorniest.  Nonetheless, in order to decouple the tests from the implementation could you add a case where the hom follows the hets and a case where the het is in the middle?

Finally, could you add a test with three hets, where the middle one is out of phase with the outer two?  This would make sure that the iteration doesn't give up somehow once it finds a phase switch.

Reply:
Added these tests, and an additional one for the scenario in which we create two phase groups from a set of haplotypes (for two pairs of in-phase variants separated by an unphaseable SNP.